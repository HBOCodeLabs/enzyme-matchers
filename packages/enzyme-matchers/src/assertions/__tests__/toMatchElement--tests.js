const { shallow, mount } = require('enzyme');
const React = require('react');

const toMatchElement = require('../toMatchElement');

function Fixture() {
  return (
    <div>
      <span id="child" className="foo" />
    </div>
  );
}

describe('toMatchElement', () => {
  const wrapper = shallow(<Fixture />);
  const truthyResults = toMatchElement(wrapper, <Fixture />);
  const truthyResults2 = toMatchElement(wrapper,
    <div>
      <span id="child" className="foo" />
    </div>
  );
  const falsyResults = toMatchElement(wrapper, <div />);
  const badInputResults = toMatchElement(mount(<Fixture />));

  it('returns the pass flag properly', () => {
    expect(truthyResults.pass).toBeTruthy();
    expect(truthyResults2.pass).toBeTruthy();
    expect(falsyResults.pass).toBeFalsy();
    expect(badInputResults.pass).toBeFalsy();
  });

  it('returns the message with the proper pass verbage', () => {
    expect(truthyResults.message).toMatchSnapshot();
    expect(truthyResults2.message).toMatchSnapshot();
    expect(falsyResults.message).toMatchSnapshot();
    expect(badInputResults.message).toMatchSnapshot();
  });

  it('returns the message with the proper fail verbage', () => {
    expect(truthyResults.negatedMessage).toMatchSnapshot();
    expect(truthyResults2.negatedMessage).toMatchSnapshot();
    expect(falsyResults.negatedMessage).toMatchSnapshot();
    expect(badInputResults.negatedMessage).toMatchSnapshot();
  });

  it('provides contextual information for the message', () => {
    expect(truthyResults.contextualInformation).toMatchSnapshot();
    expect(truthyResults2.contextualInformation).toMatchSnapshot();
    expect(falsyResults.contextualInformation).toMatchSnapshot();
    expect(badInputResults.contextualInformation).toMatchSnapshot();
  });
});
