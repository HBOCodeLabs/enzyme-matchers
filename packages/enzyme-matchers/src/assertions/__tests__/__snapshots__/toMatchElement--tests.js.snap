// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchElement provides contextual information for the message 1`] = `
"
actual:
<div>
<span id=\\"child\\" className=\\"foo\\" />
</div>

expected:
<div>
<span id=\\"child\\" className=\\"foo\\" />
</div>
"
`;

exports[`toMatchElement provides contextual information for the message 2`] = `
"
actual:
<div>
<span id=\\"child\\" className=\\"foo\\" />
</div>

expected:
<div>
<span id=\\"child\\" className=\\"foo\\" />
</div>
"
`;

exports[`toMatchElement provides contextual information for the message 3`] = `
"
actual:
<div>
<span id=\\"child\\" className=\\"foo\\" />
</div>

expected:
<div />
"
`;

exports[`toMatchElement provides contextual information for the message 4`] = `Object {}`;

exports[`toMatchElement returns the message with the proper fail verbage 1`] = `"Did not expect actual value to match the expected value."`;

exports[`toMatchElement returns the message with the proper fail verbage 2`] = `"Did not expect actual value to match the expected value."`;

exports[`toMatchElement returns the message with the proper fail verbage 3`] = `"Did not expect actual value to match the expected value."`;

exports[`toMatchElement returns the message with the proper fail verbage 4`] = `"toMatchElement must be called on a ShallowWrapper"`;

exports[`toMatchElement returns the message with the proper pass verbage 1`] = `"Expected actual value to match the expected value."`;

exports[`toMatchElement returns the message with the proper pass verbage 2`] = `"Expected actual value to match the expected value."`;

exports[`toMatchElement returns the message with the proper pass verbage 3`] = `"Expected actual value to match the expected value."`;

exports[`toMatchElement returns the message with the proper pass verbage 4`] = `"toMatchElement must be called on a ShallowWrapper"`;
